LeagueAPI/LeagueAPI/Types/LeagueId.swift

```diff
@@ -8,4 +8,4 @@
 
 import Foundation
 
-public class LeagueId: Identifier<String> {}
+public class LeagueId: Identifier<String?> {}
```

Comment:
Not a big fan at all. An identifier should never be optional.
If the leagueId is not included in certain response, then it's better to parse the leagueId field with decodeIfPresent

Reply:
decodeIfPresent returns an optional value so if it's not optional we need to set a default value for it.

![image](https://user-images.githubusercontent.com/36895261/152964896-ae14f778-f90d-4536-868e-978c80e7ef1e.png)

Reply:
You can just write
`let responseLeagueId = try container.decodeIfPresent(String.self, forKey: .leagueId)`
then assign `self.LeagueId` to `LeagueId(responseLeagueId)` if not nil, otherwise leave it nil

Reply:
If can let it empty like that : `self.leagueId = LeagueId(responseLeagueId ?? "")`
but I can't say it's nil if it's not optional